Tests/SwiftDocCTests/XCTestCase+LoadingTestData.swift

```diff
@@ -138,11 +138,14 @@ extension XCTestCase {
         var analyzer = SemanticAnalyzer(source: nil, context: context, bundle: bundle)
         let result = analyzer.visit(blockDirectiveContainer)
         
-        let problemIDs = analyzer.problems.map { problem -> String in
-            let line = problem.diagnostic.range?.lowerBound.line.description ?? "unknown-line"
+        let problemIDs = try analyzer.problems.map { problem -> (line: Int, id: String) in
```

Comment:
This doesn't feel like it's specific to this change, maybe split it up into a separate commit?

Reply:
I thought about it but it's needed for the test changes in this PR so I think it belongs here.

Some of the diagnostics being asserted on in `TabNavigatorTests` go beyond line `10` so they were being mis-sorted (`11 -` coming before `1 -`). This just resolves it by sorting on the line number specifically instead of only the string.

Reply:
OK I see, fair enough.